kuma/core/models.py

```diff
@@ -34,6 +37,21 @@ def update(self, **kw):
                                           created=False)
 
 
+class IPBan(models.Model):
+    ip = models.GenericIPAddressField()
+    created = models.DateTimeField(default=datetime.now, db_index=True)
+    deleted = models.DateTimeField(null=True, blank=True)
+
+    objects = IPBanManager()
+
+    def delete(self, *args, **kwargs):
+        self.deleted = datetime.now()
+        self.save()
```

Comment:
The 2 simple pluggable apps I've found so far are:

https://github.com/paltman/django-logicaldelete
https://github.com/hearsaycorp/django-livefield

[Others](https://www.djangopackages.com/grids/g/deletion/) seem like overkill. I'll check them out next week.